crates/experimentation_platform/src/api/experiment_groups/handlers.rs

```diff
@@ -133,8 +140,17 @@ async fn update_experiment_group(
     schema_name: SchemaName,
     user: User,
 ) -> superposition::Result<Json<ExperimentGroup>> {
-    let req = req.into_inner();
     let DbConnection(mut conn) = db_conn;
+    let experiment_group =
```

Comment:
instead of fetch and then check , we can directly write query to check if given id is of type systemgenerated